### PR TITLE
clickable light/camera gizmos

### LINKED ISSUE #8737: Picking on CameraGizmo and LightGizmo
Hello,

It would be interesting to be able to click on a gizmoLight and cameraGizmo to display the GizmoManager once clicked.

**Why ?**
This is for a publisher. The user should be able to select the GizmoCamera and GizmoLight on the stage to display the GizmoManager and move the camera or light.

It would also be useful to retrieve the light attached to the GizmoLight and retrieve the camera attached to the GizmoCamera. So that the user can modify their properties. A bit like when we select a mesh on the scene, we know that it is a mesh selected, we can retrieve the properties (position, name ...) and we can display the GizmoManager when this mesh is selected.

Would be useful if we could select (picked) the LightGizmo and CameraGizmo

Discution forum : 
https://forum.babylonjs.com/t/attachcontrol-and-cameragizmo-with-frustum/13307/5?u=dad72

I must have tried something on this PG, but it's not good when you have multiple cameraGizmo on stage. I have not been able to find or choose the correct camera, but still the last camera added to the scene. But the idea is there.
https://www.babylonjs-playground.com/#Z4ZGAY#24

Thank you in advance.

### PR DESCRIPTION
**Pull request by CedricGuillemet**
fixes #8737 
Clickable camera and light gizmos.

Example of use:
```
var gizmoManager = new BABYLON.GizmoManager(scene, 3);
gizmoManager.positionGizmoEnabled = true;

var lightGizmo = new BABYLON.LightGizmo();
lightGizmo.light = light;

lightGizmo.onClickedObservable.add((_light) => {
            manager.attachToNode(_light);              
});
```

And in a PG : https://playground.babylonjs.com/#M48IGY

### CODE REVIEW

File: src/Gizmos/lightGizmo.ts [hunk 1]
```diff
@@ -39,6 +47,19 @@ export class LightGizmo extends Gizmo {
         this._material = new StandardMaterial("light", this.gizmoLayer.utilityLayerScene);
         this._material.diffuseColor = new Color3(0.5, 0.5, 0.5);
         this._material.specularColor = new Color3(0.1, 0.1, 0.1);
+
+        if (gizmoLayer) {
```
Comment (RaananW) [inline src/Gizmos/lightGizmo.ts:51]: not needed if it is no longer optional

File: src/Gizmos/lightGizmo.ts [hunk 2]
```diff
@@ -39,6 +47,19 @@ export class LightGizmo extends Gizmo {
...
         this._material.specularColor = new Color3(0.1, 0.1, 0.1);
+
+        if (gizmoLayer) {
+            this._pointerObserver = gizmoLayer.utilityLayerScene.onPointerObservable.add((pointerInfo) => {
+                if (!this._light) {
+                    return;
+                }
+
+                var isHovered = pointerInfo.pickInfo && (this._rootMesh.getChildMeshes().indexOf(<Mesh>pointerInfo.pickInfo.pickedMesh) != -1);
+                if (isHovered && pointerInfo.type === PointerEventTypes.POINTERDOWN && pointerInfo.event.button === 0) {
```
Comment (RaananW) [inline src/Gizmos/lightGizmo.ts:58]: If this should be executed only on pointer down you can prefilter using the second variable of the `add` function:

```javascript
gizmoLayer.utilityLayerScene.onPointerObservable.add((pointerInfo) => {
    /* guaranteed to be a pointer down event*/
}, PointerEventType.POINTERDOWN);
```

File: src/Gizmos/lightGizmo.ts [hunk 3]
```diff
@@ -25,12 +27,18 @@ export class LightGizmo extends Gizmo {
     private _cachedPosition = new Vector3();
     private _cachedForward = new Vector3(0, 0, 1);
     private _attachedMeshParent: TransformNode;
+    private _pointerObserver: Nullable<Observer<PointerInfo>> = null;
+
+    /**
+     * Event that fires each time the gizmo is clicked
+     */
+    public onClickedObservable = new Observable<Light>();
```
Comment (RaananW) [inline src/Gizmos/lightGizmo.ts:35]: Don't forget to clear this on dispose()

File: src/Gizmos/cameraGizmo.ts [hunk 4]
```diff
@@ -19,17 +21,36 @@ export class CameraGizmo extends Gizmo {
     private _cameraMesh: Mesh;
     private _cameraLinesMesh: Mesh;
     private _material: StandardMaterial;
+    private _pointerObserver: Nullable<Observer<PointerInfo>> = null;
+
+    /**
+     * Event that fires each time the gizmo is clicked
+     */
+    public onClickedObservable = new Observable<Camera>();
```
Comment (RaananW) [inline src/Gizmos/cameraGizmo.ts:29]: clear on dispose

File: src/Gizmos/cameraGizmo.ts [hunk 5]
```diff
@@ -19,17 +21,36 @@ export class CameraGizmo extends Gizmo {
...
-    constructor(gizmoLayer?: UtilityLayerRenderer) {
+    constructor(gizmoLayer: UtilityLayerRenderer = UtilityLayerRenderer.DefaultUtilityLayer) {
         super(gizmoLayer);
 
         this._material = new StandardMaterial("cameraGizmoMaterial", this.gizmoLayer.utilityLayerScene);
         this._material.diffuseColor = new Color3(0.5, 0.5, 0.5);
         this._material.specularColor = new Color3(0.1, 0.1, 0.1);
+
+        if (gizmoLayer) {
+            this._pointerObserver = gizmoLayer.utilityLayerScene.onPointerObservable.add((pointerInfo) => {
```
Comment (RaananW) [inline src/Gizmos/cameraGizmo.ts:43]: same prefilter as in lightGizmo can be applied here

**Review by RaananW — CHANGES_REQUESTED** [review]
the only real change is clearing the observable(s)  when disposing. the rest are just suggestions :-)

File: src/Gizmos/cameraGizmo.ts (hunk 5 above)
Comment (CedricGuillemet) [reply src/Gizmos/cameraGizmo.ts]: Fixed

File: src/Gizmos/cameraGizmo.ts (hunk 4 above)
Comment (CedricGuillemet) [reply src/Gizmos/cameraGizmo.ts:29]: Fixed

File: src/Gizmos/lightGizmo.ts (hunk 3 above)
Comment (CedricGuillemet) [reply src/Gizmos/lightGizmo.ts:35]: Fixed

File: src/Gizmos/lightGizmo.ts (hunk 2 above)
Comment (CedricGuillemet) [reply src/Gizmos/lightGizmo.ts]: Fixed

File: src/Gizmos/lightGizmo.ts (hunk 1 above)
Comment (CedricGuillemet) [reply src/Gizmos/lightGizmo.ts]: Fixed

**Review by RaananW — APPROVED** [review]

**Pull request merged**